store/postgres/src/deployment_store.rs

```diff
@@ -1444,3 +1470,43 @@ impl DeploymentStore {
             .await
     }
 }
+
+/// Tries to fetch a [`Table`] either by its Entity name or its SQL name.
+///
+/// Since we allow our input to be either camel-case or snake-case, we must retry the
+/// search using the latter if the search for the former fails.
+fn resolve_table_name<'a>(layout: &'a Layout, name: &'_ str) -> Result<&'a Table, StoreError> {
+    layout
+        .table_for_entity(&EntityType::new(name.to_owned()))
+        .map(Deref::deref)
+        .or_else(|_error| {
+            let sql_name = SqlName::from(name);
+            layout
+                .table(&sql_name)
+                .ok_or_else(|| StoreError::UnknownTable(name.to_owned()))
+        })
+}
+
+// Resolves column names.
+//
+// Since we allow our input to be either camel-case or snake-case, we must retry the
+// search using the latter if the search for the former fails.
+fn resolve_column_names<'a, T: AsRef<str>>(
+    table: &'a Table,
+    field_names: &[T],
+) -> Result<Vec<&'a str>, StoreError> {
+    field_names
+        .iter()
+        .map(|f| {
+            table
+                .column_for_field(f.as_ref())
+                .or_else(|_error| {
+                    let sql_name = SqlName::from(f.as_ref());
+                    table
+                        .column(&sql_name)
+                        .ok_or_else(|| StoreError::UnknownField(f.as_ref().to_string()))
+                })
+                .map(|column| column.name.as_str())
+        })
+        .collect()
+}
```

Comment:
I'm not sure if those two free functions should be methods of `Layout` and `Table,` respectively.

Both types use different signatures for lookups depending if the search term uses GraphQ or SQL notation:
- `Layout`:
  - https://github.com/graphprotocol/graph-node/blob/6eb80ad561a579183f902fccfb3cb1826f8855c5/store/postgres/src/relational.rs#L507
  - https://github.com/graphprotocol/graph-node/blob/6eb80ad561a579183f902fccfb3cb1826f8855c5/store/postgres/src/relational.rs#L514
  
- `Table`
  - https://github.com/graphprotocol/graph-node/blob/6eb80ad561a579183f902fccfb3cb1826f8855c5/store/postgres/src/relational.rs#L1182
  - https://github.com/graphprotocol/graph-node/blob/6eb80ad561a579183f902fccfb3cb1826f8855c5/store/postgres/src/relational.rs#L1194

If necessary, I can turn them into methods and, maybe, homogenize those returns values.